cmd/trace-agent/subcommands/run/command.go

```diff
@@ -49,40 +41,25 @@ func setParamFlags(cmd *cobra.Command, cliParams *RunParams) {
 	cmd.PersistentFlags().StringVarP(&cliParams.CPUProfile, "cpu-profile", "l", "",
 		"enables CPU profiling and specifies profile path.")
 	cmd.PersistentFlags().StringVarP(&cliParams.MemProfile, "mem-profile", "m", "",
-		"enables memory profiling and specifies profilh.")
+		"enables memory profiling and specifies profile.")
 
 	setOSSpecificParamFlags(cmd, cliParams)
 }
 
-func runFx(ctx context.Context, cliParams *RunParams, defaultConfPath string) error {
+func runFx(cliParams *RunParams, defaultConfPath string) error {
 	if cliParams.ConfPath == "" {
 		cliParams.ConfPath = defaultConfPath
 	}
-	return fxutil.OneShot(Run,
-		fx.Supply(&contextSupplier{ctx: ctx}),
-		fx.Supply(cliParams),
-		config.Module,
+	return fxutil.Run(
```

Comment:
Is it expected to not call `Run` function (`fxutil.OneShot(Run`)?  Can you explain why you changed from `fxutil.OneShot` to ` fxutil.Run`?

Reply:
@ogaca-dd I expected to be more idiomatic - in `fx` ecosystem - to encapsulate everything in the component, and rely in `fx.Run` to initialize everything.

Reply:
`fxutil.Run` is used only in [systray](https://github.com/DataDog/datadog-agent/blob/972c3b86fd2bffbf615b31b55db2f5fffba6879a/cmd/systray/command/command.go#L82) and unclear why `fxutil.OneShot` is not used. `fxutil.OneShot` is used at more than 100 locations. So except if there is a specific reason to use `fx.Run`, I think be consistent with the rest of the code is important. What do you think?

Reply:
@ogaca-dd Sure, I prefer to be consistent. I'll rework it to use `fxutil.OneShot` again.

Reply:
Sorry, I had another look and it is fine to use `fxutil.Run`.